internal/resources/customiamrole/schema.go

```diff
@@ -0,0 +1,229 @@
+				Default:     false,
+				Optional:    true,
+			},
+		},
+	},
+}
+
+var AllowedScopesSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: fmt.Sprintf("The allowed scopes for the iam role.\nValid values are (%s)", strings.Join(AllowedScopesValidValues, ", ")),
+	Required:    true,
+	Elem: &schema.Schema{
+		Type:             schema.TypeString,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(AllowedScopesValidValues, false)),
+	},
+}
+
+var AggregationRuleSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Aggregation rules for the iam role.",
+	MaxItems:    1,
+	Optional:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			ClusterRoleSelectorKey: {
+				Type:        schema.TypeList,
+				Description: "Cluster role selector for the iam role.",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						MatchLabelsKey: {
+							Type:        schema.TypeMap,
+							Description: "Map of {key,value} pairs.\nA single {key,value} in the match_labels map is equivalent to an element of match_expression, whose key field is \"key\", the operator is \"In\", and the values array contains only \"value\". \nThe requirements are ANDed.",
+							Optional:    true,
+						},
+						MatchExpressionKey: {
+							Type:        schema.TypeList,
+							Description: "List of label selector requirements.\nThe requirements are ANDed.",
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									MeKey: {
+										Type:        schema.TypeString,
+										Description: "Key is the label key that the selector applies to.",
+										Required:    true,
+									},
+									MeOperatorKey: {
+										Type:        schema.TypeString,
+										Description: "Operator represents a key's relationship to a set of values.\nValid operators are \"In\", \"NotIn\", \"Exists\" and \"DoesNotExist\".",
+										Required:    true,
+									},
+									MeValuesKey: {
+										Type:        schema.TypeList,
+										Description: "Values is an array of string values.\nIf the operator is \"In\" or \"NotIn\", the values array must be non-empty.\nIf the operator is \"Exists\" or \"DoesNotExist\", the values array must be empty.\nThis array is replaced during a strategic merge patch.",
+										Optional:    true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+var KubernetesPermissionsSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Kubernetes permissions for the iam role.",
+	MaxItems:    1,
+	Optional:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			RuleKey: RuleSchema,
+		},
+	},
+}
+
+var RuleSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Kubernetes rules.",
+	Required:    true,
```

Comment:
NonResourceUrls are also part of the Kubernetes rules, was that omitted intentionally?